local-deploy.sh

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env bash
+
+# Version Control Center - The following variables are updated to specify the dependency versions Fortis is currentl running on.
+#       For Example: minJava is the minimum Java version required at time of publish. Currently it is 1.7.0 which this script removes decimals
+#       for arithmetic analysis when validating your machine's Jdk version. - Updated 10.25.2017
+minJava=170
+minMaven=300
+minNode=400
+minNpm=300
+minScala=270
+minCassandra=300
+minSpark=2
+#----------------------------------------------------------------------------------
+echo "Checking your current system for Fortis compatability:"
+echo
+echo "Validating JAVA version and jdk...........................-"
+
+if type -p java; then
+    echo found JAVA executable in PATH
+    _java=java
+elif [[ -n "$JAVA_HOME" ]] && [[ -x "$JAVA_HOME/bin/java" ]];  then
+    echo PASS: found JAVA executable in JAVA_HOME     
+    _java="$JAVA_HOME/bin/java"
+else
+    echo "JAVA was not found on your machine"
+fi
+
+if [[ "$_java" ]]; then
+    version=$("$_java" -version 2>&1 | awk -F '"' '/version/ {print $2}')
+    vCompare_java=$(echo "${version}" | sed -e 's/[._-]//g'| awk '{print substr($0,0,3)}')
+    echo version: "$version"
+    if (( $(echo "$vCompare_java" "$minJava" | awk '{print ($1 > $2)}') )); then
+        echo PASS: JAVA version is 1.8 or greater
+    else         
+        echo WARN: JAVA version is less than 1.8
+    fi
+fi
+
+if [[ "$_java" ]]; then
+    versionJ=$("$_java" -version 2>&1 | awk -F '"' '/version/ {print $1}')
+    echo your machine is running: "$versionJ"
+    if [[ "$versionJ" == "java version " ]]; then
+        echo WARN: java version is running the incorrect jdk package, we recommend installing openJDK
+    elif [[ "$versionJ" == "openjdk version " ]]; then
+        echo PASS: you are running the correct openJDK version
+    else         
+        echo WARN: detected unknown jdk package, please install openJDK
+    fi
+fi
+echo "Validating jdk...........................................-"
+if type -p javac; then
+    echo PASS: found javac executable in PATH
+elif [[ -n "$JAVAC_HOME" ]] && [[ -x "$JAVAC_HOME/bin/javac" ]];  then
+    echo PASS: found javac executable in JAVAC_HOME     
+else
+    echo FAIL: "javac was not found on your machine"
+fi
+#----------------------------------------------------------------------------------
+
+echo
+echo "Validating maven..........................................-"
+
+if type -p mvn; then
+    echo PASS: found maven executable in PATH
+    _mvn=mvn
+elif [[ -n "$MAVEN_HOME" ]] && [[ -x "$MAVEN_HOME/bin/mvn" ]];  then
+    echo found mvn executable in MAVEN_HOME     
+else
+    echo "FAIL: maven was not found on your system"
+fi
+
+if [[ "$_mvn" ]]; then
+    version=$("$_mvn" -version 1>&1 | awk -F '"' '/Apache/ {print $0}')
+    vCompare_mvn=$(echo "${version}" 2>&1 | sed -e 's/[A-Za-z ._-]//g'| awk '{print substr($0,0,3)}')
```

Comment:
This version number parsing is happening in a few places; would it be worth extracting this to a method?

```sh
to_major_version() { echo "$1" | sed -e 's/[A-Za-z ._-]//g'| awk '{print substr($0,0,3)}'); }
```